ganttproject/src/net/sourceforge/ganttproject/action/resource/AssignmentToggleAction.java

```diff
@@ -0,0 +1,92 @@
+Copyright (C) 2019 GanttProject Team
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 3
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package net.sourceforge.ganttproject.action.resource;
+
+import net.sourceforge.ganttproject.action.GPAction;
+import net.sourceforge.ganttproject.gui.UIFacade;
+import net.sourceforge.ganttproject.gui.UIFacade.Choice;
+import net.sourceforge.ganttproject.resource.AssignmentContext;
+import net.sourceforge.ganttproject.resource.HumanResource;
+import net.sourceforge.ganttproject.task.ResourceAssignment;
+import net.sourceforge.ganttproject.task.ResourceAssignmentMutator;
+import net.sourceforge.ganttproject.task.Task;
+import net.sourceforge.ganttproject.util.StringUtils;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Action that adds/removes the assignment of a task to a resource
+ */
+public class AssignmentToggleAction extends GPAction {
+  private final HumanResource myHumanResource;
+
+  private final UIFacade myUIFacade;
+
+  private final ResourceAssignmentMutator myMutator;
+  private final Task myTask;
+  private ArrayList<ResourceAssignment> myAssignments;
+
+  public AssignmentToggleAction(HumanResource hr, UIFacade uiFacade, Task task) {
+    super(hr.getName(), IconSize.TOOLBAR_SMALL);
+    myHumanResource = hr;
+    myUIFacade = uiFacade;
+    myTask = task;
+    myMutator = task.getAssignmentCollection().createMutator();
+    myAssignments = new ArrayList<ResourceAssignment>(Arrays.asList(myTask.getAssignmentCollection().getAssignments()));
+  }
+
+  private void createAssignment() {
+      ResourceAssignment newAssignment = myMutator.addAssignment(myHumanResource);
+      newAssignment.setLoad(100);
+
+      boolean coord = false;
+      if (myAssignments.isEmpty())
+        coord = true;
+      newAssignment.setCoordinator(coord);
+      newAssignment.setRoleForAssignment(newAssignment.getResource().getRole());
+      myAssignments.add(newAssignment);
+  }
+
+    public void delete(ResourceAssignment ra) {
+                ra.delete();
+                myAssignments.remove(ra);
+    }
+
+  @Override
+  public void actionPerformed(ActionEvent e) {
+      boolean exists = false;
+      for (ResourceAssignment ra: myAssignments) {
+          if (ra.getResource().equals(myHumanResource)){
```

Comment:
So this checks if it is delete or create action. However, you already did check it when you created the action. `getValue(Action.SELECTED_KEY)` will return you true if resource is assigned ( => this is delete action)

Reply:
Indeed, I added the Action.SELECTED_KEY logic after this code and then I forgot to refactor. Fixed